File: javascript/bob/bob.js
var Bob = function() {};

function is_shouting(input) {
    for (const c of input) {
        if (c != c.toUpperCase()) {
            return false;
        }
    }
    return input.match(/[a-z]/i);
}

Bob.prototype.hey = function(input) {
    if (is_shouting(input)) {
        return 'Whoa, chill out!';
    } else if (input[input.length - 1] == '?') {
        return 'Sure.';
    } else if (!input.trim()) {
        return 'Fine. Be that way!';
    }
    return 'Whatever.';
};

module.exports = Bob;

File: scala/bob/src/main/scala/Bob.scala
class Bob {
  def hey(s: String): String =
    if (s.exists(_.isLetter) && s.forall(c => !c.isLetter || c.isUpper))
      "Whoa, chill out!"
    else if (s.endsWith("?"))
      "Sure."
    else if (s.trim.isEmpty)
      "Fine. Be that way!"
    else
      "Whatever."
}
